src/main/src/main/java/org/geoserver/config/GeoServerReinitializer.java

```diff
@@ -0,0 +1,13 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.config;
+
+
+/**
+ * Extension point interface for initializing based on configuration.
+ * Intended to be run both upon configuration initialization, and upon reload
+ *
+ */
+public interface GeoServerReinitializer extends GeoServerInitializer { }
```

Comment:
Could I suggest adding a {{reinit}} method here, you can use a default implementation that just calls {{init}}. That way you have the flexibility to perform a different action on reload.